src/synapse.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Mock implementation of the main Synapse class
+ */
+
+import type { Synapse as ISynapse, SynapseOptions, StorageOptions, TokenAmount } from './types.js'
+import { MockStorageService } from './storage-service.js'
+
+export class MockSynapse implements ISynapse {
+  private _options: SynapseOptions
+  private _balance: number = 100 // Mock starting balance
+
+  constructor(options: SynapseOptions) {
+    this._options = options
+    console.log('[Mock] Synapse initialized with options:', {
+      withCDN: options.withCDN,
+      rpcAPI: options.rpcAPI,
+      serviceContract: options.serviceContract,
+    })
+  }
+
+  async balance(): Promise<TokenAmount> {
+    console.log('[Mock] Checking balance...')
+
+    // Simulate network delay
+    await new Promise(resolve => setTimeout(resolve, 200))
+
+    return this._balance
+  }
+
+  async deposit(amount: TokenAmount): Promise<TokenAmount> {
+    const depositAmount = typeof amount === 'string' ? parseFloat(amount) : Number(amount)
+    console.log(`[Mock] Depositing ${depositAmount} USDFC...`)
+
+    // Simulate network delay
+    await new Promise(resolve => setTimeout(resolve, 1000))
+
+    this._balance += depositAmount
```

Comment:
are you concerned about the balance getting out of sync with network state? Would it be cleaner to always sync it and not cache it?

Reply:
Downside: Read after write consistency might be sacrificed, and that's usually more important to keep up, for DX

Reply:
no, `this._balance` doesn't exist in the current form of this code, it's silly to have it locally